### PR TITLE
Add support for all create team parameters

### PR DESCRIPTION
**Pull request by timja**
# Description 
Add support for all create team parameters

Fixes https://github.com/github-api/github-api/issues/455

Implements all parameters from here: https://developer.github.com/v3/teams/#parameters

# Before submitting a PR:
We love getting PRs, but we hate asking people for the same basic changes every time. 

- [x] Push your changes to a branch other than `master`. Create your PR from that branch.    
- [x] Add JavaDocs and other comments
- [x] Write tests that run and pass in CI. See [CONTRIBUTING.md](CONTRIBUTING.md) for details on how to capture snapshot data.
- [x] Run `mvn -P ci install site` locally. This may reformat your code, commit those changes. If this command doesn't succeed, your change will not pass CI.

### CODE REVIEW

File: src/main/java/org/kohsuke/github/GHTeam.java [hunk 1]
```diff
@@ -106,6 +125,18 @@ public void setDescription(String description) throws IOException {
...
+    /**
+     * Sets privacy.
+     *
+     * @param privacy
+     *            the privacy
+     * @throws IOException
+     *             the io exception
+     */
+    public void setPrivacy(Privacy privacy) throws IOException {
+        root.createRequest().method("PATCH").with("privacy", privacy).withUrlPath(api("")).send();
```
Comment (bitwiseman) [inline src/main/java/org/kohsuke/github/GHTeam.java]: @timja 
This matches the existing behavior of other set methods on this class. There have been issues filed describing this is not great.  

This is not a blocker for this PR, but I'd like your thoughts on #589.

File: src/main/java/org/kohsuke/github/GHOrganization.java [hunk 2]
```diff
@@ -455,6 +455,21 @@ public GHTeam createTeam(String name, GHRepository... repositories) throws IOExc
...
+     *
+     * <p>
+     * You use the returned builder to set various properties, then call {@link GHCreateTeamBuilder#create()} to finally
+     * create a team.
+     *
+     * @param name
+     *            the name
+     * @return the gh create repository builder
+     */
+    public GHCreateTeamBuilder createTeam(String name) {
```
Comment (bitwiseman) [inline src/main/java/org/kohsuke/github/GHOrganization.java]: I think the other `createTeam` methods should be marked as `@Deprecated`.  Do you agree?

File: src/main/java/org/kohsuke/github/GHCreateTeamBuilder.java [hunk 3]
```diff
@@ -0,0 +1,92 @@
+package org.kohsuke.github;
+
+import java.io.IOException;
+import java.util.List;
+
+/**
+ * Creates a team.
```
Comment (bitwiseman) [inline src/main/java/org/kohsuke/github/GHCreateTeamBuilder.java]: Add a link to the github api documentation for create team

File: src/main/java/org/kohsuke/github/GHOrganization.java [hunk 4]
```diff
@@ -455,6 +455,21 @@ public GHTeam createTeam(String name, GHRepository... repositories) throws IOExc
...
+        return new GHCreateTeamBuilder(root, "/orgs/" + login + "/teams", name);
```
Comment (bitwiseman) [inline src/main/java/org/kohsuke/github/GHOrganization.java]:
```suggestion
        return new GHCreateTeamBuilder(root, login, name);
```

Or maybe even this (and add name to builder):
```suggestion
        return new GHCreateTeamBuilder(root);
```
Is a team always created attached to a login?  As me, could I create a team that your login owns?

File: src/main/java/org/kohsuke/github/GHTeam.java (hunk 1 above)
Comment (timja) [reply src/main/java/org/kohsuke/github/GHTeam.java]: agree, changed to update.
I didn't like the name but I followed the other method for consistency in the class

File: src/main/java/org/kohsuke/github/GHOrganization.java (hunk 4 above)
Comment (timja) [reply src/main/java/org/kohsuke/github/GHOrganization.java]: I did this for consistency with the create repo builder, I've made it take an org name now and not the 'apiTailUrl'.

I've kept the name as it's the only mandatory parameter for the builder

File: src/main/java/org/kohsuke/github/GHCreateTeamBuilder.java (hunk 3 above)
Comment (timja) [reply src/main/java/org/kohsuke/github/GHCreateTeamBuilder.java]: Done

File: src/main/java/org/kohsuke/github/GHOrganization.java (hunk 2 above)
Comment (timja) [reply src/main/java/org/kohsuke/github/GHOrganization.java]: Yes, wanted your input on it first before I did it, done.

File: src/main/java/org/kohsuke/github/GHCreateTeamBuilder.java [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+ */
+public class GHCreateTeamBuilder {
```
[the remark this answers is not in the document]
Comment (timja) [reply src/main/java/org/kohsuke/github/GHCreateTeamBuilder.java]: Done

File: src/main/java/org/kohsuke/github/GHTeam.java (hunk 1 above)
Comment (bitwiseman) [reply src/main/java/org/kohsuke/github/GHTeam.java]: Glad you agree. But leave this for now. I don't want to do this unevenly (some update some set).  😄

File: src/main/java/org/kohsuke/github/GHTeamBuilder.java [hunk 6]
```diff
@@ -5,16 +5,18 @@
 
 /**
  * Creates a team.
+ *
+ * https://developer.github.com/v3/teams/#parameters
```
Comment (bitwiseman) [inline src/main/java/org/kohsuke/github/GHTeamBuilder.java]:
```suggestion
 * https://developer.github.com/v3/teams/#create-team
```

File: src/main/java/org/kohsuke/github/GHTeam.java [hunk 7]
```diff
@@ -126,14 +126,14 @@ public void setDescription(String description) throws IOException {
...
-     * Sets privacy.
+     * Updates the team's privacy setting.
      *
      * @param privacy
      *            the privacy
      * @throws IOException
      *             the io exception
      */
-    public void setPrivacy(Privacy privacy) throws IOException {
+    public void updatePrivacy(Privacy privacy) throws IOException {
```
Comment (bitwiseman) [inline src/main/java/org/kohsuke/github/GHTeam.java]:
```suggestion
    public void setPrivacy(Privacy privacy) throws IOException {
```

File: src/main/java/org/kohsuke/github/GHTeamBuilder.java [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+    }
+
+    /**
+     * Maintainers for this team.
+     *
+     * @param maintainers
+     *            maintainers of team
+     * @return a builder to continue with building
+     */
+    public GHTeamBuilder maintainers(List<String> maintainers) {
```
Comment (bitwiseman) [inline src/main/java/org/kohsuke/github/GHTeamBuilder.java]: Question: 
```suggestion
    public GHTeamBuilder maintainers(String... maintainers) {
```

File: src/main/java/org/kohsuke/github/GHTeamBuilder.java [hunk 9]
```diff
@@ -0,0 +1,94 @@
...
+    }
+
+    /**
+     * Repo names to add this team to.
+     *
+     * @param repoNames
+     *            repoNames to add team to
+     * @return a builder to continue with building
+     */
+    public GHTeamBuilder repoNames(List<String> repoNames) {
```
Comment (bitwiseman) [inline src/main/java/org/kohsuke/github/GHTeamBuilder.java]: Same question:
```suggestion
    public GHTeamBuilder repositories(GHRepository... repositories) {
```

File: src/main/java/org/kohsuke/github/GHTeamBuilder.java (hunk 8 above)
Comment (bitwiseman) [reply src/main/java/org/kohsuke/github/GHTeamBuilder.java]: I mean these suggestions as added convenience methods. ... But maybe they can wait to a future release.

Comment (timja) [reply src/main/java/org/kohsuke/github/GHTeamBuilder.java]: makes sense, no annoying singletonList to use it

**Pull request merged**